jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/DynamicUtils.java

```diff
@@ -55,195 +55,262 @@
                     subProcessNodeInstance.setProcessInstance(realProcessInstance);
                     if (dynamicContext == null) {
-                    	subProcessNodeInstance.setNodeInstanceContainer(realProcessInstance);
+                        subProcessNodeInstance.setNodeInstanceContainer(realProcessInstance);
                     } else {
-	                    DynamicNodeInstance realDynamicContext = findDynamicContext(realProcessInstance, dynamicContext.getUniqueId());
-	                    subProcessNodeInstance.setNodeInstanceContainer(realDynamicContext);
+                        DynamicNodeInstance realDynamicContext = findDynamicContext(realProcessInstance,
+                                                                                    dynamicContext.getUniqueId());
+                        subProcessNodeInstance.setNodeInstanceContainer(realDynamicContext);
                     }
-                    return executeSubProcess((StatefulKnowledgeSessionImpl) ksession, processId, parameters, processInstance, subProcessNodeInstance);
+                    return executeSubProcess((StatefulKnowledgeSessionImpl) ksession,
+                                             processId,
+                                             parameters,
+                                             processInstance,
+                                             subProcessNodeInstance);
                 }
             });
-    	} else {
-    		throw new IllegalArgumentException("Unsupported ksession: " + ksession == null ? "null" : ksession.getClass().getName());
-    	}
-	}
-	
-	private static long executeSubProcess(StatefulKnowledgeSessionImpl ksession, String processId, Map<String, Object> parameters, ProcessInstance processInstance, SubProcessNodeInstance subProcessNodeInstance) {
-		Process process = ksession.getKieBase().getProcess(processId);
+        } else {
+            throw new IllegalArgumentException("Unsupported ksession: " + ksession == null ? "null" : ksession.getClass().getName());
+        }
+    }
+
+    private static long executeSubProcess(StatefulKnowledgeSessionImpl ksession,
+                                          String processId,
+                                          Map<String, Object> parameters,
+                                          ProcessInstance processInstance,
+                                          SubProcessNodeInstance subProcessNodeInstance) {
+        Process process = ksession.getKieBase().getProcess(processId);
         if (process == null) {
-            logger.error("Could not find process {}", processId);
+            logger.error("Could not find process {}",
+                         processId);
             logger.error("Aborting process");
-        	processInstance.setState(ProcessInstance.STATE_ABORTED);
-        	
-        	return -1;
+            processInstance.setState(ProcessInstance.STATE_ABORTED);
```

Comment:
@mswiderski when debugging this i noticed that  setting the process state to aborted is i think even wrong. imo user error should not abort it in this case. wdyt?